libs/community/langchain_community/vectorstores/yellowbrick.py

```diff
@@ -46,67 +56,144 @@ def __init__(
         """
 
         import psycopg2
+        from psycopg2 import extras
+
+        extras.register_uuid()
 
         if not isinstance(embedding, Embeddings):
             warnings.warn("embeddings input must be Embeddings object.")
 
         self.connection_string = connection_string
         self._table = table
         self._embedding = embedding
-        self._connection = psycopg2.connect(connection_string)
+        self._max_embedding_len = None
+        self.idle_threshold_seconds = idle_threshold_seconds
+        self._connection = psycopg2.connect(self.connection_string)
+        self.last_used_time = time.time()
+        self._check_database_utf8()
 
-        self.__post_init__()
+        if drop:
+            self._drop_table(self._table)
+            self._drop_table(self._table + self.CONTENT_TABLE)
+            self._drop_lsh_index_tables()
 
-    def __post_init__(
-        self,
-    ) -> None:
-        """Initialize the store."""
-        self.check_database_utf8()
-        self.create_table_if_not_exists()
+        self._create_table()
+
+        self._use_lsh = use_lsh
+        self._hamming_distance = 0
+
+        if seed is not None:
+            random.seed(seed)
+        self._seed = seed
 
     def __del__(self) -> None:
         if self._connection:
             self._connection.close()
 
-    def create_table_if_not_exists(self) -> None:
+    @property
+    def use_lsh(self) -> bool:
+        return self._use_lsh
+
+    @use_lsh.setter
+    def use_lsh(self, use_lsh: bool) -> None:
+        self._use_lsh = use_lsh
+        if self._use_lsh:
+            self._create_lsh_index_tables()
+
+    @property
+    def hamming_distance(self) -> bool:
+        return self._hamming_distance
+
+    @hamming_distance.setter
+    def hamming_distance(self, hamming_distance: int) -> None:
+        self._hamming_distance = hamming_distance
+
+    def _get_connection(self):
```

Comment:
I think it would be useful to introduce a new method that provides a context manager for the cursor so that cursor can be closed properly on exception

```
from contextlib import contextmanager

@contetmanager
def get_cursor():
   with connection... as conn: # <-- Can do this or the logic that re-uses the connection etc
      with conn.cusor() as cursor: 
        yield cursor
```

Reply:
Good suggestion. I've implemented a context manager.